iroh-net/src/netcheck/reportgen.rs

```diff
@@ -253,6 +252,7 @@ impl Actor {
 
                 _ = &mut probe_timer => {
                     warn!("probes timed out");
+                    probes.abort_all();
```

Comment:
The probes are aborted on line 303 and they get a log message there.  if you abort here i'm not sure if you get the log message.  (admittedly the log message there is a bit misleading in case of a timeout, maybe it could be tweaked)

Then `self.handle_abort_probes()` makes sure that we break out of the loop.  It's designed this way because it should not matter whether we fall in this condition by the timeout or by the actor message, and this way everything is done by the same code.

Reply:
removed